datafusion/expr/src/udf.rs

```diff
@@ -649,13 +649,31 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
 
     /// Calculates the [`SortProperties`] of this function based on its
     /// children's properties.
-    fn output_ordering(&self, _inputs: &[ExprProperties]) -> Result<SortProperties> {
-        Ok(SortProperties::Unordered)
+    fn output_ordering(&self, inputs: &[ExprProperties]) -> Result<SortProperties> {
+        if !self.output_preserves_lex_ordering(inputs)? {
+            return Ok(SortProperties::Unordered);
+        }
+
+        let Some(first_order) = inputs.first().map(|p| &p.sort_properties) else {
+            return Ok(SortProperties::Unordered);
+        };
+
+        if inputs.iter().skip(1).all(|input| &input.sort_properties == first_order) {
+            Ok(first_order.clone())
+        } else {
+            Ok(SortProperties::Unordered)
+        }
     }
 
     /// Whether the function preserves lexicographical ordering based on the input ordering
-    fn output_preserves_lex_ordering(&self, _inputs: &[ExprProperties]) -> Result<bool> {
-        Ok(false)
+    fn output_preserves_lex_ordering(&self, inputs: &[ExprProperties]) -> Result<bool> {
```

Comment:
Isn't it just returning false?

Reply:
> Isn't it just returning false?

No, the function doesn't simply return false. It returns true when there's at least one input and all inputs share identical sort properties. I've refactored the code to make this logic more readable.

Reply:
But the condition you've said is checked already at the output_ordering() part. This output_preserves_lex_ordering() is only for labeling the functions which preserves the lex ordering. You define it as false here, and only preserver functions override it as false. So, this API will not be open to outside, and the users still interact with the output_ordering() API. Am I missing smth?

Reply:
Yes, makes sense now. Should be resolved in 542e1dc.